docs/contribute/plugin/plugin_requirement.md

```diff
@@ -0,0 +1,336 @@
+---
+sidebar_position: 2
+---
+
+# Requirements for Developing WasmEdge Plugin
+
+By developing a plugin, one can extend the functionality of WasmEdge and customize it to suit the specific needs. Here is a flowchart showing all the steps needed for developing WasmEdge Plugin -
+
+```mermaid
+graph TD;
+A(Choose programming language) --> B(Install WasmEdge)
+A --> C(Build WasmEdge from source)
+A --> D(Install Rust)
+B --> E(Set up development environment)
+C --> E
+D --> E
+E --> F(Create project directory)
+E --> G(Add configuration files)
+E --> H(Install necessary tools and dependencies)
+E --> I(Enable specific backends or components)
+E --> J(Write plugin code)
+E --> K(Build plugin)
+F --> L(Define plugin API)
+K --> L
+L --> M(Compile WasmEdge plugin)
+M --> N(Test and debug plugin)
+
+```
+
+This flowchart illustrates the process of developing a WasmEdge plugin, showcasing the steps involved from choosing a programming language to finalizing and releasing the plugin.
+
+> For existing plugins, please refer to their respective [documentation pages](/contribute/source/plugin) for specific instructions.
+
+## Choose programming language
+
+The first step in developing a WasmEdge plugin is to choose the programming language one wants to use. Currently, C, C++, and Rust are supported by the WasmEdge runtime.
+
+- **C:** Developers who choose to develop WasmEdge plugins using the C API need to install WasmEdge before starting the development process. Detailed instructions on installing WasmEdge can be found in the [install WasmEdge](/develop/build-and-run/install.md) guide.
+  
+- **C++:** For developers interested in developing WasmEdge plugins using the internal C++ API, they should first build WasmEdge from source. The process for building WasmEdge from source is outlined in th [build WasmEdge from source](../source/build_from_src.md) before starting.Once WasmEdge is successfully built, they can proceed with developing plugins using the internal C++ API.
```

Comment:
Why we don't need to install WasmEdge here?

Reply:
I got the info from the existing documentation of Develop WasmEdge Plug-in in C++ API
 - https://wasmedge.org/docs/contribute/plugin/develop_plugin_cpp#prerequisites